From 3b3b1ee360d1ae4515b7c8cb6aca7bccea7a74cd Mon Sep 17 00:00:00 2001
From: Igal Zeltser <izeltser@habana.ai>
Date: Tue, 25 May 2021 14:10:06 +0300
Subject: [PATCH] [SW-43147] i2cslave: support core voltage command

Support i2c slave direct command '113-114' which returns
the core voltage of the device.

Change-Id: I67974c2586def3b644d26bca9fae32abee9108a1
Signed-off-by: Igal Zeltser <izeltser@habana.ai>
---

diff --git a/preboot/module_i2c_slave.c b/preboot/module_i2c_slave.c
index e1bbf43..68a2797 100644
--- a/preboot/module_i2c_slave.c
+++ b/preboot/module_i2c_slave.c
@@ -41,6 +41,18 @@
 							max_onchip_temp;
 }
 
+static u8 i2cslave_core_voltage_cmd(u8 cmd)
+{
+	switch (cmd) {
+	case HL_VOLTAGES_ASIC_CORE_LOW:
+		return (u8)HL_ASIC_CORE_VOLTAGE;
+	case HL_VOLTAGES_ASIC_CORE_HIGH:
+		return (u8)(HL_ASIC_CORE_VOLTAGE >> 8);
+	default:
+		return 0;
+	}
+}
+
 u8 i2cslave_handle_read_req(u8 cmd, u32 read_seq)
 {
 	u8 ret;
@@ -127,6 +139,10 @@
 	case HL_MODULE_TDP_LEVEL:
 		ret = HL_MODULE_TDP_LEVEL_1;
 		break;
+	case HL_VOLTAGES_ASIC_CORE_LOW:
+	case HL_VOLTAGES_ASIC_CORE_HIGH:
+		ret = i2cslave_core_voltage_cmd(cmd);
+		break;
 	default:
 		if (HL_I2C_EVENT_IS_PEC(cmd))
 			ret = i2cslave_pec;
